backend/drm/atomic.c

```diff
@@ -140,16 +140,19 @@ static bool atomic_conn_enable(struct wlr_drm_backend *drm,
 		return !enable;
 	}
 
+	/* crtc will be enabled on next page flip */
+	if (enable) {
+		crtc->needs_modeset = true;
+		return true;
+	}
+
 	struct atomic atom;
 	atomic_begin(crtc, &atom);
 	atomic_add(&atom, crtc->id, crtc->props.active, enable);
-	if (enable) {
-		atomic_add(&atom, conn->id, conn->props.crtc_id, crtc->id);
-		atomic_add(&atom, crtc->id, crtc->props.mode_id, crtc->mode_id);
-	} else {
-		atomic_add(&atom, conn->id, conn->props.crtc_id, 0);
-		atomic_add(&atom, crtc->id, crtc->props.mode_id, 0);
-	}
+	atomic_add(&atom, conn->id, conn->props.crtc_id, 0);
```

Comment:
This disables the plane even if `enable == true`.

Reply:
no, we return in early in the `enable` case, see above.

Reply:
Hmm. So we don't commit anymore. If we don't commit, we don't know whether we can enable the output (bandwidth limits).

Reply:
but isn't this only really useful if we also assign plane props, etc. which happens on pageflip (which is where it happens now)?

Reply:
Is there another solution i'm missing? it'd be nice to have these drm drivers also work out of the box with wlroots.